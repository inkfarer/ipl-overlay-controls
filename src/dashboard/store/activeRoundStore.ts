import { ActiveRound, SwapColorsInternally } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { GameWinner } from 'types/enums/gameWinner';
import { SetActiveColorRequest, SetWinnerRequest, SwapRoundColorRequest } from 'types/messages/activeRound';
import { SetRoundRequest } from 'types/messages/rounds';
import { ActiveRoundGame } from 'types/activeRoundGame';

const activeRound = nodecg.Replicant<ActiveRound>('activeRound');
const swapColorsInternally = nodecg.Replicant<SwapColorsInternally>('swapColorsInternally');

export const activeRoundReps = [ activeRound, swapColorsInternally ];

export interface ActiveRoundStore {
    activeRound: ActiveRound;
    swapColorsInternally: SwapColorsInternally;
}

export const activeRoundStore = createStore<ActiveRoundStore>({
    state: {
        activeRound: null,
        swapColorsInternally: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        setWinner(store, { winner }: { winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner });
        },
        removeWinner(): void {
            nodecg.sendMessage('removeWinner');
        },
        setActiveColor(store, request: SetActiveColorRequest): void {
            nodecg.sendMessage('setActiveColor', request);
        },
        swapColors(): void {
            swapColorsInternally.value = !swapColorsInternally.value;
        },
        setWinnerForIndex(store, { index, winner }: { index: number, winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner, roundIndex: index } as SetWinnerRequest);
        },
        resetActiveRound(): void {
            nodecg.sendMessage('resetActiveRound');
        },
        setActiveRound(store, request: SetRoundRequest): void {
            nodecg.sendMessage('setActiveRound', request);
        },
        swapRoundColor(store, request: SwapRoundColorRequest): void {
            nodecg.sendMessage('swapRoundColor', request);
        },
        updateActiveGames(store, games: ActiveRoundGame[]): void {
            nodecg.sendMessage('updateActiveGames', { games });
        },
        switchToNextColor(): void {
            nodecg.sendMessage('switchToNextColor');
        },
        switchToPreviousColor(): void {
            nodecg.sendMessage('switchToPreviousColor');
        },
        async getNextAndPreviousColors(): Promise<void> {
            return nodecg.sendMessage('getNextAndPreviousColors');
        }
    }
});

export const activeRoundStoreKey: InjectionKey<Store<ActiveRoundStore>> = Symbol();

export function useActiveRoundStore(): Store<ActiveRoundStore> {
    return useStore(activeRoundStoreKey);
}
