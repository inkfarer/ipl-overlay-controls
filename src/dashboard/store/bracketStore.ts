import { Bracket } from '@tourneyview/common';
import { defineStore } from 'pinia';

const bracketData = nodecg.Replicant<Bracket | undefined>('bracketData');

export const bracketReps = [ bracketData ];

export interface BracketStore {
    bracketData: Bracket | undefined;
}

export const useBracketStore = defineStore('brackets', {
    state: () => ({
        bracketData: undefined
    } as BracketStore),
    actions: {
        resetBracketData() {
            bracketData.value = undefined;
        }
    }
});
