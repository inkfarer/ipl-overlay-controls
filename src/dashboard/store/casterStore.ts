import { NodeCGBrowser } from 'nodecg/browser';
import { Caster, Casters, RadiaSettings } from 'schemas';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

const casters = nodecg.Replicant<Casters>('casters');
const radiaSettings = nodecg.Replicant<RadiaSettings>('radiaSettings');

export const castersReps = [ casters, radiaSettings ];

export interface CasterStore {
    casters: Casters;
    uncommittedCasters: Casters;
    radiaSettings: RadiaSettings;
}

const defaultCaster = {
    name: 'New Caster',
    twitter: '',
    pronouns: '?/?'
};

export const useCasterStore = defineStore('casters', {
    state: () => ({
        casters: {},
        uncommittedCasters: {},
        radiaSettings: {
            enabled: null,
            guildID: null,
            updateOnImport: null
        }
    } as CasterStore),
    actions: {
        updateCaster({ id, newValue }: { id: string, newValue: Caster }): void {
            casters.value[id] = newValue;
        },
        removeUncommittedCaster(id: string): void {
            delete this.uncommittedCasters[id];
        },
        addUncommittedCaster({ id, caster }: { id: string, caster?: Caster }): void {
            this.uncommittedCasters[id] = caster ?? defaultCaster;
        },
        addDefaultCaster(): string {
            const id = generateId();
            this.addUncommittedCaster({ id });
            return id;
        },
        async saveUncommittedCaster({ id, caster }: { id: string, caster: Caster }): Promise<string> {
            const newId = await nodecg.sendMessage('saveCaster', caster);
            this.removeUncommittedCaster(id);
            return newId;
        },
        async loadCastersFromVc(): Promise<void> {
            const result = await nodecg.sendMessage('getLiveCommentators');
            if (result.extra && result.extra.length > 0) {
                for (let i = 0; i < result.extra.length; i++) {
                    const extraCaster = result.extra[i];
                    const id = extraCaster.discord_user_id;
                    delete extraCaster.discord_user_id;
                    this.addUncommittedCaster({ id, caster: extraCaster });
                }
            }
        },
        showCasters() {
            nodecg.sendMessage('mainShowCasters');
        },
        async removeCaster(id: string): Promise<void> {
            return nodecg.sendMessage('removeCaster', { id });
        }
    }
});
