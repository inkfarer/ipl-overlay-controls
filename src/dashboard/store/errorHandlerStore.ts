import { App } from 'vue';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

export const useErrorHandlerStore = defineStore('errorHandler', {
    state: () => ({
        recentErrors: {}
    }),
    actions: {
        removeRecentError({ key }: { key: string }): void {
            delete this.recentErrors[key];
        },
        handleError({ err, info }: { err: unknown, info: string }): void {
            console.error(`Got error from '${info}': \n`, err);

            if (Object.keys(this.recentErrors).length >= 2) return;

            const id = generateId();
            this.recentErrors[id] = err;
            window.setTimeout(() => {
                delete this.recentErrors[id];
            }, 25000);
        }
    }
});

export function setUpErrorHandler(app: App<unknown>): void {
    const store = useErrorHandlerStore();
    app.config.errorHandler = (err, vm, info) => {
        store.handleError({ err, info });
    };
}
