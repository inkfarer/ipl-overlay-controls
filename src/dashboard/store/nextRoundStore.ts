import { NextRound } from 'schemas';
import { SetNextRoundRequest } from 'types/messages/rounds';
import { BeginNextMatchRequest } from 'types/messages/activeRound';
import { defineStore } from 'pinia';

const nextRound = nodecg.Replicant<NextRound>('nextRound');

export const nextRoundReps = [ nextRound ];

export interface NextRoundStore {
    nextRound: NextRound;
}

export const useNextRoundStore = defineStore('nextRound', {
    state: () => ({
        nextRound: null
    } as NextRoundStore),
    actions: {
        setShowOnStream(newValue: boolean) {
            nextRound.value.showOnStream = newValue;
        },
        beginNextMatch(request: BeginNextMatchRequest): void {
            nodecg.sendMessage('beginNextMatch', request);
        },
        setNextRound(request: SetNextRoundRequest): void {
            nodecg.sendMessage('setNextRound', request);
        }
    }
});
