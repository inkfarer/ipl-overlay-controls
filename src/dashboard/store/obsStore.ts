import { NodeCGBrowser } from 'nodecg/browser';
import { ObsCredentials, ObsData } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { SetObsDataRequest } from 'types/messages/obs';

const obsData = nodecg.Replicant<ObsData>('obsData');
const obsCredentials = nodecg.Replicant<ObsCredentials>('obsCredentials');

export const obsReps = [obsData, obsCredentials];

export interface ObsStore {
    obsData: ObsData
    obsCredentials: ObsCredentials
}

export const obsStore = createStore<ObsStore>({
    state: {
        obsData: null,
        obsCredentials: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        async connect(store, { address, password }: { address: string, password?: string }): Promise<void> {
            return nodecg.sendMessage('connectToObs', { address, password });
        },
        async setData(store, data: SetObsDataRequest): Promise<void> {
            return nodecg.sendMessage('setObsData', data);
        },
        async startGame(): Promise<void> {
            return nodecg.sendMessage('startGame');
        },
        async endGame(): Promise<void> {
            return nodecg.sendMessage('endGame');
        },
        setEnabled(store, enabled: boolean): void {
            nodecg.sendMessage('setObsSocketEnabled', enabled);
        }
    }
});

export const obsStoreKey: InjectionKey<Store<ObsStore>> = Symbol();

export function useObsStore(): Store<ObsStore> {
    return useStore(obsStoreKey);
}
