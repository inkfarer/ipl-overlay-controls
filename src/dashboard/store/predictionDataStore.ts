import { NodeCGBrowser } from 'nodecg/browser';
import { Prediction, PredictionStore } from 'schemas';
import { PredictionStatus } from 'types/enums/predictionStatus';
import { defineStore } from 'pinia';

const predictionStore = nodecg.Replicant<PredictionStore>('predictionStore');

export const predictionReps = [ predictionStore ];

export interface PredictionDataStore {
    predictionStore: PredictionStore;
}

export const usePredictionDataStore = defineStore('prediction', {
    state: () => ({
        predictionStore: null,
    } as PredictionDataStore),
    actions: {
        async lockPrediction() {
            return patchPrediction(
                this.predictionStore.currentPrediction,
                PredictionStatus.LOCKED,
                [ PredictionStatus.ACTIVE ]);
        },
        async cancelPrediction() {
            return patchPrediction(
                this.predictionStore.currentPrediction,
                PredictionStatus.CANCELED,
                [ PredictionStatus.ACTIVE, PredictionStatus.LOCKED ]);
        },
        async resolvePrediction({ winningOutcomeIndex }: { winningOutcomeIndex: number }) {
            if (winningOutcomeIndex > 1 || winningOutcomeIndex < 0) {
                throw new Error(`Cannot resolve prediction with outcome index ${winningOutcomeIndex}.`);
            } else if (!this.predictionStore.currentPrediction?.id) {
                throw new Error('No prediction to resolve.');
            } else if (this.predictionStore.currentPrediction.status !== PredictionStatus.LOCKED ) {
                throw new Error('Can only resolve a locked prediction.');
            }

            return nodecg.sendMessage('patchPrediction', {
                id: this.predictionStore.currentPrediction.id,
                status: PredictionStatus.RESOLVED,
                winning_outcome_id: this.predictionStore.currentPrediction.outcomes[winningOutcomeIndex].id
            });
        },
        async createPrediction(
            { title, teamAName, teamBName, duration }:
                { title: string, teamAName: string, teamBName: string, duration: number }
        ) {
            if (
                [PredictionStatus.ACTIVE, PredictionStatus.LOCKED]
                    .includes(this.predictionStore.currentPrediction?.status as PredictionStatus)
            ) {
                throw new Error('An unresolved prediction already exists.');
            }

            return nodecg.sendMessage('postPrediction', {
                title,
                outcomes: [
                    { title: teamAName },
                    { title: teamBName }
                ],
                prediction_window: duration
            });
        },
        async reconnect() {
            return nodecg.sendMessage('reconnectToRadiaSocket');
        }
    }
});

async function patchPrediction(
    currentPrediction: Prediction,
    status: PredictionStatus,
    allowedForStatuses: PredictionStatus[]
): Promise<void> {
    if (!currentPrediction || !allowedForStatuses.includes(currentPrediction?.status as PredictionStatus)) {
        throw new Error(`Cannot set prediction status at this time. Status must match one of [${allowedForStatuses.join(', ')}]`);
    }

    return nodecg.sendMessage('patchPrediction', {
        id: currentPrediction.id,
        status: status,
    });
}
