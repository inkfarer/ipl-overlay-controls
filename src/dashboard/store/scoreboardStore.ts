import { NodeCGBrowser } from 'nodecg/browser';
import { ScoreboardData } from 'schemas';
import { defineStore } from 'pinia';

const scoreboardData = nodecg.Replicant<ScoreboardData>('scoreboardData');

export const scoreboardReps = [scoreboardData];

export interface ScoreboardStore {
    scoreboardData: ScoreboardData
}

export const useScoreboardStore = defineStore('scoreboard', {
    state: () => ({
        scoreboardData: {
            flavorText: '',
            isVisible: null
        }
    } as ScoreboardStore),
    actions: {
        setFlavorText(flavorText: string): void {
            scoreboardData.value.flavorText = flavorText;
        },
        setScoreboardVisible(isVisible: boolean): void {
            scoreboardData.value.isVisible = isVisible;
        }
    }
});
