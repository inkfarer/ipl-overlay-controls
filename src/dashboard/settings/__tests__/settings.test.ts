import { config, mount } from '@vue/test-utils';
import Settings from '../settings.vue';
import { createStore } from 'vuex';
import { obsStoreKey } from '../../store/obsStore';

describe('Settings', () => {
    config.global.stubs = {
        IplErrorDisplay: true,
        LastfmSettings: true,
        RadiaSettings: true,
        ObsSocketSettings: true,
        ObsDataPicker: true,
        RuntimeConfig: true,
        FontAwesomeIcon: true
    };

    function createObsStore() {
        return createStore({
            state: {
                obsData: {
                    enabled: true
                }
            }
        });
    }

    it.each(['lastfm', 'radia', 'gameVersion', 'obs-socket'])('matches snapshot when section %s is selected', async (section) => {
        const obsStore = createObsStore();
        const wrapper = mount(Settings, {
            global: {
                plugins: [[obsStore, obsStoreKey]]
            }
        });

        wrapper.getComponent('[data-test="open-sidebar-button"]').vm.$emit('click');
        await wrapper.vm.$nextTick();
        wrapper.getComponent(`[data-test="section-selector_${section}"]`).vm.$emit('click');
        await wrapper.vm.$nextTick();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when OBS socket is disabled', async () => {
        const obsStore = createObsStore();
        obsStore.state.obsData.enabled = false;
        const wrapper = mount(Settings, {
            global: {
                plugins: [[obsStore, obsStoreKey]]
            }
        });

        wrapper.getComponent('[data-test="open-sidebar-button"]').vm.$emit('click');
        await wrapper.vm.$nextTick();
        wrapper.getComponent('[data-test="section-selector_obs-socket"]').vm.$emit('click');
        await wrapper.vm.$nextTick();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
