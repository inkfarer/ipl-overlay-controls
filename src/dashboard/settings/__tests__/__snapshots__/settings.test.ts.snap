// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings matches snapshot when OBS socket is disabled 1`] = `
"<ipl-error-display-stub class=\\"m-b-8\\"></ipl-error-display-stub>
<div class=\\"ipl-sidebar__wrapper sidebar-transition-enter-from sidebar-transition-leave-from sidebar-transition-leave-active\\" data-v-2b6fb510=\\"\\">
  <div class=\\"background\\" data-v-2b6fb510=\\"\\"></div>
  <div class=\\"content\\" data-v-2b6fb510=\\"\\">
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_lastfm\\" data-v-11cc5f1c=\\"\\">Last.fm</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_radia\\" data-v-11cc5f1c=\\"\\">Radia</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_gameVersion\\" data-v-11cc5f1c=\\"\\">Game version</div>
    <div class=\\"ipl-space color-blue clickable m-b-8\\" data-test=\\"section-selector_obs-socket\\" data-v-11cc5f1c=\\"\\">OBS Socket</div>
  </div>
</div>
<!---->
<div class=\\"ipl-space color-dark clickable layout horizontal m-t-8\\" data-test=\\"open-sidebar-button\\" data-v-11cc5f1c=\\"\\">
  <font-awesome-icon-stub icon=\\"bars\\" border=\\"false\\" fixedwidth=\\"false\\" listitem=\\"false\\" pulse=\\"false\\" swapopacity=\\"false\\" spin=\\"false\\" symbol=\\"false\\" inverse=\\"false\\" class=\\"large-icon m-r-8\\"></font-awesome-icon-stub> All Settings
</div>
<obs-socket-settings-stub class=\\"m-t-8\\"></obs-socket-settings-stub>
<!--v-if-->"
`;

exports[`Settings matches snapshot when section gameVersion is selected 1`] = `
"<ipl-error-display-stub class=\\"m-b-8\\"></ipl-error-display-stub>
<div class=\\"ipl-sidebar__wrapper sidebar-transition-enter-from sidebar-transition-leave-from sidebar-transition-leave-active\\" data-v-2b6fb510=\\"\\">
  <div class=\\"background\\" data-v-2b6fb510=\\"\\"></div>
  <div class=\\"content\\" data-v-2b6fb510=\\"\\">
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_lastfm\\" data-v-11cc5f1c=\\"\\">Last.fm</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_radia\\" data-v-11cc5f1c=\\"\\">Radia</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_gameVersion\\" data-v-11cc5f1c=\\"\\">Game version</div>
    <div class=\\"ipl-space color-blue clickable m-b-8\\" data-test=\\"section-selector_obs-socket\\" data-v-11cc5f1c=\\"\\">OBS Socket</div>
  </div>
</div>
<!---->
<div class=\\"ipl-space color-dark clickable layout horizontal m-t-8\\" data-test=\\"open-sidebar-button\\" data-v-11cc5f1c=\\"\\">
  <font-awesome-icon-stub icon=\\"bars\\" border=\\"false\\" fixedwidth=\\"false\\" listitem=\\"false\\" pulse=\\"false\\" swapopacity=\\"false\\" spin=\\"false\\" symbol=\\"false\\" inverse=\\"false\\" class=\\"large-icon m-r-8\\"></font-awesome-icon-stub> All Settings
</div>
<runtime-config-stub class=\\"m-t-8\\"></runtime-config-stub>"
`;

exports[`Settings matches snapshot when section lastfm is selected 1`] = `
"<ipl-error-display-stub class=\\"m-b-8\\"></ipl-error-display-stub>
<div class=\\"ipl-sidebar__wrapper sidebar-transition-enter-from sidebar-transition-leave-from sidebar-transition-leave-active\\" data-v-2b6fb510=\\"\\">
  <div class=\\"background\\" data-v-2b6fb510=\\"\\"></div>
  <div class=\\"content\\" data-v-2b6fb510=\\"\\">
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_lastfm\\" data-v-11cc5f1c=\\"\\">Last.fm</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_radia\\" data-v-11cc5f1c=\\"\\">Radia</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_gameVersion\\" data-v-11cc5f1c=\\"\\">Game version</div>
    <div class=\\"ipl-space color-blue clickable m-b-8\\" data-test=\\"section-selector_obs-socket\\" data-v-11cc5f1c=\\"\\">OBS Socket</div>
  </div>
</div>
<!---->
<div class=\\"ipl-space color-dark clickable layout horizontal m-t-8\\" data-test=\\"open-sidebar-button\\" data-v-11cc5f1c=\\"\\">
  <font-awesome-icon-stub icon=\\"bars\\" border=\\"false\\" fixedwidth=\\"false\\" listitem=\\"false\\" pulse=\\"false\\" swapopacity=\\"false\\" spin=\\"false\\" symbol=\\"false\\" inverse=\\"false\\" class=\\"large-icon m-r-8\\"></font-awesome-icon-stub> All Settings
</div>
<lastfm-settings-stub class=\\"m-t-8\\"></lastfm-settings-stub>"
`;

exports[`Settings matches snapshot when section obs-socket is selected 1`] = `
"<ipl-error-display-stub class=\\"m-b-8\\"></ipl-error-display-stub>
<div class=\\"ipl-sidebar__wrapper sidebar-transition-enter-from sidebar-transition-leave-from sidebar-transition-leave-active\\" data-v-2b6fb510=\\"\\">
  <div class=\\"background\\" data-v-2b6fb510=\\"\\"></div>
  <div class=\\"content\\" data-v-2b6fb510=\\"\\">
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_lastfm\\" data-v-11cc5f1c=\\"\\">Last.fm</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_radia\\" data-v-11cc5f1c=\\"\\">Radia</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_gameVersion\\" data-v-11cc5f1c=\\"\\">Game version</div>
    <div class=\\"ipl-space color-blue clickable m-b-8\\" data-test=\\"section-selector_obs-socket\\" data-v-11cc5f1c=\\"\\">OBS Socket</div>
  </div>
</div>
<!---->
<div class=\\"ipl-space color-dark clickable layout horizontal m-t-8\\" data-test=\\"open-sidebar-button\\" data-v-11cc5f1c=\\"\\">
  <font-awesome-icon-stub icon=\\"bars\\" border=\\"false\\" fixedwidth=\\"false\\" listitem=\\"false\\" pulse=\\"false\\" swapopacity=\\"false\\" spin=\\"false\\" symbol=\\"false\\" inverse=\\"false\\" class=\\"large-icon m-r-8\\"></font-awesome-icon-stub> All Settings
</div>
<obs-socket-settings-stub class=\\"m-t-8\\"></obs-socket-settings-stub>
<obs-data-picker-stub class=\\"m-t-8\\"></obs-data-picker-stub>"
`;

exports[`Settings matches snapshot when section radia is selected 1`] = `
"<ipl-error-display-stub class=\\"m-b-8\\"></ipl-error-display-stub>
<div class=\\"ipl-sidebar__wrapper sidebar-transition-enter-from sidebar-transition-leave-from sidebar-transition-leave-active\\" data-v-2b6fb510=\\"\\">
  <div class=\\"background\\" data-v-2b6fb510=\\"\\"></div>
  <div class=\\"content\\" data-v-2b6fb510=\\"\\">
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_lastfm\\" data-v-11cc5f1c=\\"\\">Last.fm</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_radia\\" data-v-11cc5f1c=\\"\\">Radia</div>
    <div class=\\"ipl-space color-light clickable m-b-8\\" data-test=\\"section-selector_gameVersion\\" data-v-11cc5f1c=\\"\\">Game version</div>
    <div class=\\"ipl-space color-blue clickable m-b-8\\" data-test=\\"section-selector_obs-socket\\" data-v-11cc5f1c=\\"\\">OBS Socket</div>
  </div>
</div>
<!---->
<div class=\\"ipl-space color-dark clickable layout horizontal m-t-8\\" data-test=\\"open-sidebar-button\\" data-v-11cc5f1c=\\"\\">
  <font-awesome-icon-stub icon=\\"bars\\" border=\\"false\\" fixedwidth=\\"false\\" listitem=\\"false\\" pulse=\\"false\\" swapopacity=\\"false\\" spin=\\"false\\" symbol=\\"false\\" inverse=\\"false\\" class=\\"large-icon m-r-8\\"></font-awesome-icon-stub> All Settings
</div>
<radia-settings-stub class=\\"m-t-8\\"></radia-settings-stub>"
`;
